api/src/main/java/org/eclipse/microprofile/openapi/models/media/Schema.java

```diff
@@ -1192,4 +1256,928 @@ default Schema oneOf(List<Schema> oneOf) {
+     *            a list of example objects which this schema could describe
+     * @return current Schema instance
+     * @since 4.0
+     */
+    default Schema examples(List<Object> examples) {
+        setExamples(examples);
+        return this;
+    }
+
+    /**
+     * Adds an example to the examples list.
+     *
+     * @param example
+     *            the example to add to the examples list
+     * @return current Schema instance
+     * @since 4.0
+     */
+    Schema addExample(Object example);
+
+    /**
+     * Removes an example from the examples list.
+     *
+     * @param example
+     *            the example to remove from the examples list
+     * @since 4.0
+     */
+    void removeExample(Object example);
+
+    /**
+     * Gets a schema property by name.
+     * <p>
+     * Allows access to arbitrary properties in a schema object, allowing use of alternative schema dialects which use
+     * different property names (or the same property names with different data types).
+     * <p>
+     * When using the standard schema dialect, this method can be used to retrieve values set by other methods. E.g.
+     *
+     * <pre>
+     * {@code
+     * schema.setMinimum(new BigDecimal(3));
+     * BigDecimal minimum = (BigDecimal) schema.get("minimum"); // returns 3
+     * }
+     * </pre>
+     *
+     * @param propertyName
+     *            the property name
+     * @return the value of the named property, or {@code null} if a property with the given name is not set
+     */
+    Object get(String propertyName);
+
+    /**
+     * Sets a schema property.
+     * <p>
+     * Allows the modifications of arbitrary schema properties in a schema properties, allowing use of alternative
+     * schema dialects which use different property names (or the same property names with different data types).
+     * <p>
+     * Passing {@code null} as the {@code value} removes the property from the schema object.
+     * <p>
+     * {@code value} must be one of the following types, otherwise non-portable behavior results:
+     * <ul>
+     * <li>Any primitive type
+     * <li>Any primitive wrapper class
+     * <li>{@code null}
+     * <li>{@code String}
+     * <li>{@code BigDecimal}
+     * <li>{@code BigInteger}
+     * <li>Any type which {@link OASFactory} can create
+     * <li>Any Enumeration
+     * <li>{@code List<T>} where {@code T} is a permitted type
+     * <li>{@code Map<String, T>} where {@code T} is a permitted type
+     * </ul>
+     *
+     * <p>
+     * When using the standard schema dialect, values set by this method can be retrieved by other methods. E.g.
+     *
+     * <pre>
+     * {@code
+     * schema.set("minimum", new BigDecimal(3));
+     * BigDecimal minimum = schema.getMinimum(); // returns 3
+     * }
+     * </pre>
+     *
+     * @param propertyName
+     *            the property name
+     * @param the
+     *            value to set
+     */
+    void set(String propertyName, Object value);
```

Comment:
Need to decide on the semantics for `get` and `set` when used with a collection or map.

In general, getters for a collection return a "copy (potentially immutable)", so that you can't get a collection, modify it and have the modification reflected in the model. Setters make no indication of whether they they copy the argument or not. Users are expected to either replace the whole collection, or use the relevant `add` and `remove` methods to modify the collection indirectly.

However, getters returning other model objects do not copy them. Similarly, getters returning `Object` don't copy them, even if the object happens to be a collection.

Given that the primary use case is to get and set arbitrary properties, I think it makes most sense to follow the same semantics for properties where we accept `Object` so I'm leaning towards not requiring a copy here.

Reply:
I agree that `get` and `set` can be non-copying operations.